sdjournal/journal.go

```diff
@@ -84,13 +260,50 @@ func (m *Match) String() string {
 	return m.Field + "=" + m.Value
 }
 
+func (j *Journal) getFunction(name string) (unsafe.Pointer, error) {
+	j.mu.Lock()
+	defer j.mu.Unlock()
+	f, ok := libsystemdFunctions[name]
+	if !ok {
+		var err error
+		f, err = j.lib.GetSymbolPointer(name)
+		if err != nil {
+			return nil, err
+		}
+
+		libsystemdFunctions[name] = f
+	}
+
+	return f, nil
+}
+
 // NewJournal returns a new Journal instance pointing to the local journal
-func NewJournal() (*Journal, error) {
-	j := &Journal{}
-	r := C.sd_journal_open(&j.cjournal, C.SD_JOURNAL_LOCAL_ONLY)
+func NewJournal() (j *Journal, err error) {
+	h, err := dlopen.GetHandle(libsystemdNames)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
```

Comment:
Yeah, but if we close it and errors before closing it will leak and the user will see only the first error... Not really such a big deal but I think it's different from the file case.